dirty_cat/tests/test_deduplicate.py

```diff
@@ -0,0 +1,124 @@
+import string
+from typing import List
+
+import numpy as np
+import pandas as pd
+import pytest
+from scipy.cluster.hierarchy import linkage
+from scipy.spatial.distance import squareform
+
+from dirty_cat._deduplicate import (
+    _compute_ngram_distance,
+    _create_spelling_correction,
+    _guess_clusters,
+    deduplicate,
+)
+
+
+def generate_example_data(examples, entries_per_example, prob_mistake_per_letter, rng):
+    """Helper function to generate data consisting of multiple entries per example.
+    Characters are misspelled with probability `prob_mistake_per_letter`"""
+
+    data = []
+    for example, n_ex in zip(examples, entries_per_example):
+        len_ex = len(example)
+        # generate a 2D array of chars of size (n_ex, len_ex)
+        str_as_list = np.array([list(example)] * n_ex)
+        # randomly choose which characters are misspelled
+        idxes = np.where(
+            rng.random_sample(len(example[0]) * n_ex) < prob_mistake_per_letter
+        )[0]
+        # and randomly pick with which character to replace
+        replacements = [
+            string.ascii_lowercase[i]
+            for i in rng.choice(np.arange(26), len(idxes)).astype(int)
+        ]
+        # introduce spelling mistakes at right examples and char locations per example
+        str_as_list[idxes // len_ex, idxes % len_ex] = replacements
+        # go back to 1d array of strings
+        data.append(np.ascontiguousarray(str_as_list).view(f"U{len_ex}").ravel())
+    return np.concatenate(data)
```

Comment:
Import this function from datasets, see comment in example section.

Reply:
I left the function, because it's slightly different - for ease of testing I pass the randomstate explicitly here, but for the user facing function in `datasets` this makes it unncecessarily complicated IMO.
If you prefer to make it DRY-er, I can add the randomstate argument to the function in `datasets` as well.

Reply:
Fine for me. I think the most important was to make the example lighter. If this one is different you can keep it in the tests.